content/post/2022-02-21-code-style/index.Rmd

```diff
@@ -0,0 +1,173 @@
+-   [Google R Style guide](https://google.github.io/styleguide/Rguide.html) with e.g. a preference for BigCamelCase for function names.
+-   [The style guide of the mlr organziation (for machine learning in R)](https://github.com/mlr-org/mlr3/wiki/Style-Guide).
+-   Your own preferences? Yes it's fine to have some as long as your team agrees. :wink: Feel free to mention your preferences in the comments. This is a non-judgmental space (or indent :grin:).
+
+Most style guides will have some preferences regarding code spacing, or "breathing".
+
+`r hugodown::embed_tweet("1504499938302046214")`
+
+`r hugodown::embed_tweet("1486396341542481922")`
+
+Another excellent resource is [Jenny Bryan's useR! 2018 keynote *"Code Smells and Feels"*](https://github.com/jennybc/code-smells-and-feels).
+It's more focused on code structure and commenting.
+
+### General resources
+
+The resources listed below are books without a free version online.
+Hopefully libraries and used book stores can help.
+
+`r hugodown::embed_tweet("951152160435310592")`
+
+-   [The Art of Readable Code by Dustin Boswell, Trevor Foucher](https://www.goodreads.com/book/show/8677004-the-art-of-readable-code) can be (even for those who do not view it as a beach read :cocktail:), a short, light and actionable read.
+
+-   [Refactoring by Martin Fowler](https://www.goodreads.com/book/show/44936.Refactoring) is an inspiration for the aforementioned keynote talk by Jenny Bryan.
+    It defines code smells and refactoring techniques.
+    See also the blogpost ["Explaining Variable" by Pete Hodgson](https://blog.thepete.net/blog/2021/06/24/explaining-variable/) (heard of via [Jenny Bryan](https://twitter.com/JennyBryan/status/1412140590842597385)).
+
+-   [The Programmer's Brain by Felienne Hermans](https://www.goodreads.com/book/show/57196550-the-programmer-s-brain) gives a perspective on e.g. how code smells or bad naming influence cognitive load.
+    It is full of practical tips.
+
+## Tools
+
+To begin with the most important, or basic, tool here is probably to use an IDE, integrated development environment, like RStudio IDE or VSCODE, as IDEs come with, or easily support, code diagnostic tools.
+
+### Diagnostic tools
+
+#### lintr
+
+Linting tools will indicate errors and potentially style preference violations.
+
+The [lintr R package by Jim Hester](https://github.com/r-lib/lintr) has a lot of useful linters such as whether code is commented, whether lines are too long etc.
+
+It is pretty common to find *linting tools* in all editors (VSCODE and others) and for most languages.
+To set up lintr with your code editor refer to [its docs](https://github.com/r-lib/lintr#editors-setup).
+Unless you have RStudio IDE and linting is the one included already in the IDE (for errors mainly only, not style preference).
+If you use VSCODE see <https://github.com/REditorSupport/vscode-R/wiki/R-Language-Service#formatting>.
```

Comment:
Formatting in the R extensions uses styler. Maybe we should move the link to part below ? 

Link for VSCODE diagnostic would be https://github.com/REditorSupport/vscode-R/wiki/R-Language-Service#diagnostics

```suggestion
If you use VSCODE, [vscode-R](https://marketplace.visualstudio.com/items?itemName=Ikuyadeu.r) extension includes diagnostic tool - see <https://github.com/REditorSupport/vscode-R/wiki/R-Language-Service#diagnostics>.
```